src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -739,113 +706,113 @@ bool IsBufferContentEqualConstantTime(const void * a, const void * b, size_t n)
-                                       MBEDTLS_ECP_PF_UNCOMPRESSED, &pubkey_size, Uint8::to_uchar(mPublicKey), mPublicKey.Length());
-    VerifyOrExit(result == 0, error = CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrExit(pubkey_size == mPublicKey.Length(), error = CHIP_ERROR_INVALID_ARGUMENT);
+    keypair->mBitLength = sKeypairBitLen;
 
-    keypair      = nullptr;
-    mInitialized = true;
+    status = psa_export_key(keyId, Uint8::to_uchar(keypair->mPrivateKey), sizeof(keypair->mPrivateKey), &privateKeyLength);
+    VerifyOrExit(status == 0, error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(privateKeyLength == sizeof(keypair->mPrivateKey), error = CHIP_ERROR_INTERNAL);
+
+    status = psa_export_public_key(keyId, Uint8::to_uchar(mPublicKey), mPublicKey.Length(), &publicKeyLength);
+    VerifyOrExit(status == 0, error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(publicKeyLength == kP256_PublicKey_Length, error = CHIP_ERROR_INTERNAL);
 
 exit:
-    if (keypair != nullptr)
+    logPsaError(status);
+    if (error == CHIP_NO_ERROR)
     {
-        mbedtls_ecp_keypair_free(keypair);
-        keypair = nullptr;
+        mInitialized = true;
     }
-
-    logMbedTLSError(result);
+    if (0 != keyId)
+    {
+        psa_destroy_key(keyId);
+    }
+    psa_reset_key_attributes(&attributes);
     return error;
 }
 
 CHIP_ERROR P256Keypair::Serialize(P256SerializedKeypair & output) const
 {
-    const mbedtls_ecp_keypair * keypair = to_const_keypair(&mKeypair);
-    size_t len                          = output.Length() == 0 ? output.Capacity() : output.Length();
+    const psa_ecp_keypair * keypair = to_const_keypair(&mKeypair);
+    size_t len                      = output.Length() == 0 ? output.Capacity() : output.Length();
     Encoding::BufferWriter bbuf(output, len);
-    uint8_t privkey[kP256_PrivateKey_Length];
     CHIP_ERROR error = CHIP_NO_ERROR;
-    int result       = 0;
 
     bbuf.Put(mPublicKey, mPublicKey.Length());
 
-    VerifyOrExit(bbuf.Available() == sizeof(privkey), error = CHIP_ERROR_INTERNAL);
-
-    VerifyOrExit(mbedtls_mpi_size(&keypair->CHIP_CRYPTO_PAL_PRIVATE(d)) <= bbuf.Available(), error = CHIP_ERROR_INTERNAL);
-    result = mbedtls_mpi_write_binary(&keypair->CHIP_CRYPTO_PAL_PRIVATE(d), Uint8::to_uchar(privkey), sizeof(privkey));
-
-    VerifyOrExit(result == 0, error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(bbuf.Available() == sizeof(keypair->mPrivateKey), error = CHIP_ERROR_INTERNAL);
 
-    bbuf.Put(privkey, sizeof(privkey));
+    bbuf.Put(keypair->mPrivateKey, PSA_BITS_TO_BYTES(keypair->mBitLength));
     VerifyOrExit(bbuf.Fit(), error = CHIP_ERROR_BUFFER_TOO_SMALL);
 
     output.SetLength(bbuf.Needed());
 
 exit:
-    ClearSecretData(privkey, sizeof(privkey));
-    logMbedTLSError(result);
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Crypto, "P256Keypair serialization error %s", error.AsString());
```

Comment:
```suggestion
        ChipLogError(Crypto, "P256Keypair serialization error " CHIP_LOG_FORMAT, error.Format());
```
This produces smaller code when verbose error messages are disabled.